### PR TITLE
[DHIS2-8695] Refactor working lists #15 - refactor working lists ancestor components

### PR DESCRIPTION
**Pull request by JoakimSM**
[DHIS2-9884 (Subtask of DHIS2-8695)](https://jira.dhis2.org/browse/DHIS2-9884)

Refactoring the ancestor components for the working lists. Adding a branch for the tracked entity instance working lists as well.

(flow error fixed in next PR)

### CODE REVIEW

File: src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/RunningMutationsHandler/WorkingListsInitRunningMutationsHandler.component.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import type { Props } from './WorkingListsInitRunningMutationsHandler.types';
+
+const EventWorkingListsSetupWithLoadingIndicator = withLoadingIndicator()(WorkingListsInitHeader);
+
+export const WorkingListsInitRunningMutationsHandler = ({ mutationInProgress, ...passOnProps }: Props) => (
+    <EventWorkingListsSetupWithLoadingIndicator
+        {...passOnProps}
+        ready={!mutationInProgress}
+    />
+);
```
Comment (paschalidi) [inline src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/RunningMutationsHandler/WorkingListsInitRunningMutationsHandler.component.js:14]: I could see that this exact same code could live in the `/EventWorkingListsInitConnectionStatusResolver.component.js` and we could completely delete this fie and keep the nesting more lean.

File: src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/RunningMutationsHandler/WorkingListsInitRunningMutationsHandler.types.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// @flow
+export type Props = $ReadOnly<{
+    mutationInProgress: boolean,
+}>;
```
Comment (paschalidi) [inline src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/RunningMutationsHandler/WorkingListsInitRunningMutationsHandler.types.js:4]: Annotate using `$Exact` if you feel it adds value

File: src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/Header/workingListsInitHeader.types.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// @flow
+export type Props = $ReadOnly<{}>;
```
Comment (paschalidi) [inline src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/Header/workingListsInitHeader.types.js:2]: Also here perhaps annotate using the exact thingy if you feel it worths the efford

File: src/core_modules/capture-core/components/Pages/MainPage/WorkingListsType/workingListsType.types.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+// @flow
+export type Props = $ReadOnly<{
+    programId: string,
+}>;
```
Comment (paschalidi) [inline src/core_modules/capture-core/components/Pages/MainPage/WorkingListsType/workingListsType.types.js:4]: 🏄‍♂️  going with the flow

File: src/core_modules/capture-core/components/Pages/MainPage/mainPage.types.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+type PassOnProps = $ReadOnly<{
+    programId: string,
+}>;
+
+export type Props = $ReadOnly<{
+    ...PassOnProps,
+    currentSelectionsComplete: boolean,
+    classes: Object,
+}>;
```
Comment (paschalidi) [inline src/core_modules/capture-core/components/Pages/MainPage/mainPage.types.js:11]: Just keeping track of things. Feel free to be more pragmatic and skip my suggestions

File: src/core_modules/capture-core/hooks/useProgram/useProgram.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+import { getProgramFromProgramIdThrowIfNotFound, TrackerProgram } from '../../metaData';
+import { programTypes } from './programTypes.const';
+
+export const useProgram = (programId: string) => useMemo(() => {
+    const program = getProgramFromProgramIdThrowIfNotFound(programId);
+    return {
+        program,
+        programType: program instanceof TrackerProgram ? programTypes.TRACKER_PROGRAM : programTypes.EVENT_PROGRAM,
+    };
+}, [programId]);
```
Comment (paschalidi) [inline src/core_modules/capture-core/hooks/useProgram/useProgram.js:12]: I have created a very similar hook lately but isnt yet merged. Perhaps we could align on this

**Review by paschalidi — APPROVED** [review]
Good stuff. Have some comments but you can pick if these make sense for you to add them in the codebase or not

File: src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/ConnectionStatusResolver/eventWorkingListsInitConnectionStatusResolver.types.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+// @flow
+type PassOnProps = $ReadOnly<{
+    mutationInProgress: boolean,
+}>;
+
+export type Props = $ReadOnly<{
+    ...PassOnProps,
+    isOnline: boolean,
+    listId: string,
+}>;
```
[the remark this answers is not in the document]
Comment (JoakimSM) [reply src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/ConnectionStatusResolver/eventWorkingListsInitConnectionStatusResolver.types.js]: fixed (assuming you were talking about exact annotation)

File: src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/Header/WorkingListsInitHeader.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    listContainer: {
+        padding: theme.typography.pxToRem(24),
+    },
+    title: {
+        ...theme.typography.title,
+    },
+});
+
+const WorkingListsInitHeaderPlain =
+    ({ classes: { headerContainer, listContainer, title }, ...passOnProps }: Props & CssClasses) => (
```
[the remark this answers is not in the document]
Comment (JoakimSM) [reply src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/Header/WorkingListsInitHeader.js]: 👍 Changed.

File: src/core_modules/capture-core/hooks/useProgram/useProgram.js (hunk 6 above)
Comment (JoakimSM) [reply src/core_modules/capture-core/hooks/useProgram/useProgram.js:12]: ok, good. Let's align when you merge the code.

Comment (paschalidi) [reply src/core_modules/capture-core/hooks/useProgram/useProgram.js:12]: Yep, you will see it on some of the open PRs I have. :)

File: src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/EventWorkingListsInit.container.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+// @flow
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { EventWorkingListsInitConnectionStatusResolver } from './ConnectionStatusResolver';
+
+export const EventWorkingListsInit = (props: $ReadOnly<{}>) => {
```
[the remark this answers is not in the document]
Comment (JoakimSM) [reply src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsInit/EventWorkingListsInit.container.js]: We agreed to go with `() => {}`

**Pull request merged**